pool/src/pool/actions.rs

```diff
@@ -280,6 +280,8 @@ pub fn build_actions_from_request(
             }
             _ => panic_with_error!(e, PoolError::BadRequest),
         }
+        // Verify max positions haven't been exceeded
+        pool.require_under_max(e, from_state.positions.clone(), request.request_type)
```

Comment:
We can likely get away with only checking this after all requests have been processed.

We know how many positions they started with, so if it increases, just need to make sure its at most the maximum.